addons/mail/static/tests/mock_server/mock_models/mail_message.js

```diff
@@ -200,10 +193,10 @@ export class MailMessage extends models.ServerModel {
                 Object.assign(response, { thread });
             }
             if (for_current_user) {
-                const needactionPartnerIds = allNotifications
+                const notifications_partners = allNotifications
                     .filter((notification) => !notification.is_read)
                     .map((notification) => notification.res_partner_id);
-                response["needaction_partner_ids"] = needactionPartnerIds;
+                response["needaction"] = notifications_partners.includes(this.env.user?.partner_id);
```

Comment:
Note: mock server is inconsistent, this uses notifications but some other places use value of `needaction` "field".

As a follow up, it would be best if `needaction` was removed from tests/mock server, and all code used notifications instead.